client/lib/plugins/utils.js

```diff
@@ -263,3 +265,50 @@ export function extractSearchInformation( searchTerm = '' ) {
 
 	return [ search, author ];
 }
+
+/**
+ * Returns an author keyword to be used on plugin search by author
+ * The follow actions are taken:
+ * * Try to get the main author from the list of contributors
+ * * Try to extract the author keyword from author_profile
+ * * Try to get the author_name
+ * * Send an empty string if none of the previous actions works
+ *
+ * @param plugin
+ * @returns {string} the author keyword or an empty string
+ */
+export function getPluginAuthorKeyword( plugin ) {
+	const { contributors = {} } = plugin;
+
+	return (
+		Object.keys( contributors ).find( ( contributorKey ) => {
+			const authorName = plugin.author_name;
+			const contributorName = contributors[ contributorKey ].display_name;
+
+			return (
+				authorName &&
+				contributorName &&
+				( authorName.includes( contributorName ) || contributorName.includes( authorName ) )
+			);
+		} ) ||
+		getPluginAuthorProfileKeyword( plugin ) ||
+		plugin.author_name ||
+		''
+	);
+}
+
+export const WPORG_PROFILE_URL = 'https://profiles.wordpress.org/';
+
+/**
+ * Get the author keywrod from author_profile property
+ *
+ * @param plugin
+ * @returns {string|null} the author keyword
+ */
+export function getPluginAuthorProfileKeyword( plugin ) {
+	if ( ! plugin?.author_profile?.includes( WPORG_PROFILE_URL ) ) {
```

Comment:
I realize this has alread been merged, but I think it should be fixed: theoretically, this could also match URLs like `http://malicious-website.com?somearg=https://profiles.wordpress.org/foobar`

Should we use `String.prototype.startsWith()` instead of `String.prototype.includes()`?

Reply:
Good catch, at this point I think using a Regex would be the best option for readability and to catch this kind of error.

I'll handle this in a follow-up today, changing this verification to Regex and using the "starts with" anchor (`^`).

Reply:
Thanks! 🙌 

The [String.prototype.startsWIth](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/startsWith) method is ideal when you just need to check if a string starts with another string or not.

IMHO you don't need to use a regex, you can just use `plugin?.author_profile?.startsWith( WPORG_PROFILE_URL )`. `startsWith` is faster in theory, which is why I recommend it instead of regex. 

If we talk about performance, `string.indexOf( 'somestring' ) === 0` is actually 5-10 times faster than everything else, but readability suffers from it, which is the other reason why I recommend `startsWith`.

Anyway, feel free to ping me for a review 👀

Reply:
Yes, I totally get it, it was also discussed here https://github.com/Automattic/wp-calypso/pull/60977#discussion_r807691608 due to the value returned to be `https://profiles.wordpress.org/username/` so using regex we would be able to get the username value without the two replaces ([here](https://github.com/Automattic/wp-calypso/pull/60977/files#diff-1c366daa121a1d1c948ab7cae032f09a365804b57dfa09c55b8276e6bb302247R313)).

But this is a different problem and the point here is mostly about guarantee the string starts with the WP.org URL.